lib/src/main/java/scal/io/liger/MainActivity.java

```diff
@@ -554,14 +555,26 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
 
                 if (c instanceof ClipCard) {
                     ClipCard cc = (ClipCard)c;
-                    MediaFile mf = new MediaFile(path, Constants.VIDEO);
-                    cc.saveMediaFile(mf);
 
-                    // SEEMS LIKE A REASONABLE TIME TO SAVE
-                    mStoryPathLibrary.save(true);
+                    // confirm mime type
+                    String mimeType = URLConnection.guessContentTypeFromName(path);
 
-                    mCardAdapter.changeCard(cc);
-                    scrollRecyclerViewToCard(cc);
+                    Log.d(TAG, "onActivityResult, media type is " + mimeType);
+
+                    if (mimeType.startsWith(Constants.VIDEO)) {
+
+                        MediaFile mf = new MediaFile(path, Constants.VIDEO);
+                        cc.saveMediaFile(mf);
+
+                        // SEEMS LIKE A REASONABLE TIME TO SAVE
+                        mStoryPathLibrary.save(true);
+
+                        mCardAdapter.changeCard(cc);
+                        scrollRecyclerViewToCard(cc);
+                    } else {
+                        Log.e(TAG, "onActivityResult, expected " + Constants.VIDEO + " but found " + mimeType);
```

Comment:
instead of silently logging this, we need to raise the issue to the user so they know why their new import just vanished